src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -856,6 +863,109 @@ def visit_span(self, span: Span):
                 )
 
 
+class ConsecutiveDBSpanDetector(PerformanceDetector):
+    """
+    The detector first looks for X number of consecutive db query spans,
+    each being at least Y in length, where X and Y are defined in the detector threshold settings.
+    Once these set of spans are found, the detector will compare each db span in the consecutive list
+    to determine if they are dependant on one another, if they are not a performance issue is found.
+
+    This detector assuming spans are ordered chronologically
+    """
+
+    __slots__ = "stored_problems"
+
+    settings_key = DetectorType.CONSECUTIVE_DB_OP
+
+    def init(self):
+        self.stored_problems: dict[str, PerformanceProblem] = {}
+        self.consecutive_db_spans: list[Span] = []
+
+    def visit_span(self, span: Span) -> None:
+        span_id = span.get("span_id", None)
+        span_duration = get_span_duration(span)
+        exceeds_duration_threshold = span_duration > timedelta(
+            milliseconds=self.settings["duration_threshold"]
+        )
```

Comment:
For most/all of our other detectors we've been using the duration threshold to test the entire run of spans rather than individual spans. Since the problem here is "you have a run of consecutive spans that could be done in parallel" I think that's more appropriate here too. For example, if your threshold was 100ms and you had consecutive db spans of 75ms, 3ms, and 145ms, I think it would be potentially worth parallelizing those even though one of them is very short.

This doesn't matter too much while we're doing our first detector run (with `duration_threshold` set to `0`) but I think it's worth thinking about before we set that to a real value.

Reply:
Yeah, that actually makes more sense! Or another option would be to only apply the threshold to the independant spans, as those are the spans that can be in parallel instead.